imsim/sky_model.py

```diff
@@ -108,7 +113,102 @@ def __call__(self, x, y):
+    as a function of pixel coordinates relative to the value at the 
+    CCD center.
+    """
+    def __init__(self,img_wcs,c_wcs,spatial_vary):
+        self.img_wcs = img_wcs
+        self.c_wcs = c_wcs
+        self.spatial_vary = spatial_vary
+        
+    def generate_heightfield(self,fractal_dimension=2.5, n=4096):
+        '''
+        # Use the spectral sythesis method to generate a heightfield
+        '''
+        H = 1 - (fractal_dimension - 2)
+        kpow = -(H + 1.0) / 1.2
+        
+        A = np.zeros((n, n), complex)
+
+        kvec = np.fft.fftfreq(n)
+        k0 = kvec[n // 64]
+        kx, ky = np.meshgrid(kvec, kvec, sparse=True, copy=False)
+        ksq = kx ** 2 + ky ** 2
+        m = ksq > 0
+        random_seed = random.randint(1, 1000000)
+        gen = np.random.RandomState(random_seed)
+        phase = 2 * np.pi * gen.uniform(size=(n, n))
+        A[m] = ksq[m] ** kpow * gen.normal(size=(n, n))[m] * np.exp(1.j * phase[m]) * np.exp(-ksq[m] / k0 ** 2)
+
+        return np.fft.ifft2(A)
+
+
+    def simulate_fringes(self,n = 1.2,n1=1.5, amp=0.002, nwaves_rms=10.):
+        '''
+        # Generate random fringing pattern from a heightfield
+        '''
+        X = self.generate_heightfield(n, 4096)
+        X *= nwaves_rms / np.std(X.real)
+        Z =  amp * np.cos(2 * n1 * (X.real))
+        # Scale up to unity
+        Z  += 1
+        return(Z)
+
+        
+    def fringe_variation_level(self,spatial = True):
+        '''
+        Function implementing temporal and spatial variation of fringing. 
+        
+        The function will return a multiplicative factor that modifies
+        the fringing amplitude from sensor to sensor based on the 
+        time (not implemented yet) and its location on the focal plane.
+        
+        Set spatial = True to turn on spatial variation implementation.
+        Otherwise, this function will return a unity value.
+        '''
+        
+        if spatial:
+            # Load 2d interpolator for OH spatial variation
+            filename = os.path.join(data_dir, 'fringing_data',
+                                        'OH_spatial.pkl')
+            with open(filename, 'rb') as f:
+                interp = pickle.load(f)
+                
+            dx = self.c_wcs[0] - self.img_wcs.center.ra.deg 
+            dy = self.c_wcs[1] - self.img_wcs.center.dec.deg
+            
+            # calculated OH flux level wrst the center of focal plane.
+            level = interp(dx,dy)/interp(0,0)
+            
+            # Check if the angluar offset of the current sensor is outside 
+            # the coverage field of the OH data we currently have. 
+            # Set variation level to unity if True, interpolated value if False.
+            if np.isnan(level):
+                return 1
+            else:
+                return level
+        else:
+            return 1
+        
+
+    def cal_fringe_amplitude(self,x,y):
```

Comment:
Please type out the word calculate here.  Cal can be short for too many things, so better to be clear than brief.